### PR TITLE
Whitelisted variables

### PR DESCRIPTION
**Pull request by ccondra**
## Purpose of this pull request
The purpose of this PR is to update the content for the new constraints for adding variables to pages, blocks, and email templates. From the Admin, whitelisted variables must always be selected from a list, rather than entered directly to the HTML.

## Affected documentation pages

[Adding Predefined Variables](https://docs.magento.com/m2/ee/user_guide/marketing/variables-predefined.html)
- Reworked instructions
- Added new images

[Adding Custom Variables](https://docs.magento.com/m2/ee/user_guide/marketing/variables-custom.html)
- Reworked instructions
- Added different instructions for ce/ee-b2b

[Message Templates](https://docs.magento.com/m2/ee/user_guide/marketing/email-template-message.html)
- Reworked instructions
- Added different instructions for ce/ee-b2b

## Affected Magento editions

- [x] Open Source
- [x] Commerce
- [x] B2B

## Additional information

Staging #425

### CODE REVIEW

File: src/marketing/email-template-message.md [hunk 1]
```diff
@@ -2,30 +2,32 @@
 title: Message Templates
 ---
 
-The process of customizing the body of each message is the same as the customizing the header or footer. The only difference is that there is a different message template for each activity or event that triggers a notification. You can use the templates as they are, or customize them to match your voice and brand. In addition to the template text, there is a wide selection of [variables]({% link marketing/variables.md %}) that can be incorporated into the template.
+The process of customizing the body of each message is the same as the customizing the header or footer. The only difference is that there is a different message template for each activity or event that triggers a notification. You can use the templates as they are, or customize them to match your voice and brand. In addition to the template text, there is a wide selection of whitelisted [predefined]({% link marketing/variables-predefined.md %}) variables and [custom]({% link marketing/variables-custom.md %}) varibles that you can create and incorporate into the template.
```
Comment (hickskenh) [inline src/marketing/email-template-message.md:5]:
```suggestion
The process for customizing the body of each message is the same as the customizing the header or footer. The only difference is that there is a different message template for each activity or event that triggers a notification. You can use the templates as they are, or you can customize them to match your voice and brand. In addition to the template text, there is a wide selection of whitelisted [predefined]({% link marketing/variables-predefined.md %}) variables and [custom]({% link marketing/variables-custom.md %}) variables that you can create and incorporate into the template.
```

File: src/marketing/email-template-message.md [hunk 2]
```diff
@@ -2,30 +2,32 @@
...
+    - Under **Load default template**, choose the **Template** that you want to customize.
 
-    ![Email templates - load default template]({% link images/images/email-templates-message-load-default.png %}){: .zoom}
-    _Default Message Template_
+    - Click <span class="btn">Load Template</span>.
 
-## Step 2. Customize the Template
+## Step 2. Customize the template
 
 1. In the **Template Name** field, enter a name for your custom template.
```
Comment (hickskenh) [inline src/marketing/email-template-message.md:24]:
```suggestion
1. For **Template Name**, enter a name for your custom template.
```

File: src/marketing/email-template-message.md [hunk 3]
```diff
@@ -2,30 +2,32 @@
...
 
 1. If needed, change the **Template Subject**.
 
-   This is contains the first line of the message, which is the salutation by default. You can leave it as is, or enter something more descriptive.
+   This is the first line of the message, which is the salutation by default. You can leave it as is, or enter something more descriptive.
```
Comment (hickskenh) [inline src/marketing/email-template-message.md:28]:
```suggestion
   This is the first line of the message, which is the salutation by default. You can leave it as is, or you can enter something more descriptive.
```

File: src/marketing/email-template-message.md [hunk 4]
```diff
@@ -37,49 +39,56 @@ The process of customizing the body of each message is the same as the customizi
     {:.bs-callout-info}
     When working in the template code, be careful not to accidentally type over the code that is enclosed in double braces.
 
-    ![Email templates - template comtent]({% link images/images/email-template-message-content.png %}){: .zoom}
-    _Template Message Content_
+1. To insert a variable, position the cursor in the code where you want the variable to appear.
 
-1. To insert a variable, position the cursor in the code where you want the variable to appear, click <span class="btn">Insert Variable</span>, and choose the variable that you want to insert.
+   The selection of variables varies by template, and includes whitelisted predefined and custom variables, if avalable.
```
Comment (hickskenh) [inline src/marketing/email-template-message.md:44]:
```suggestion
   The selection of variables varies by template and includes whitelisted [predefined]({% link marketing/variables-predefined.md %}) and [custom]({% link marketing/variables-custom.md %}) variables, if available.
```

File: src/marketing/email-template-message.md [hunk 5]
```diff
@@ -37,49 +39,56 @@ The process of customizing the body of each message is the same as the customizi
...
+   _Insert Variable_
 
-    In addition to the Store Contact variables, the list includes the Customer Account URL, Customer Email, and Customer Name. However, you are not limited to the variables in this list. You can enter the code for any system or [custom variable]({% link marketing/variables-custom.md %}) directly into the template.
+   A command to insert the variable is enclosed in curly braces and added to the code at the cursor location. For example:
 
-1. If you need to make any CSS declarations, enter the styles in the **Template Styles** box.
+   `customVar code=my_custom_variable`
 
-## Step 3. Update the Configuration
+1. To make CSS declarations, enter the styles in the **Template Styles** box.
```
Comment (hickskenh) [inline src/marketing/email-template-message.md:55]:
```suggestion
1. To make CSS declarations, enter the styles in **Template Styles**.
```

File: src/marketing/email-template-message.md [hunk 6]
```diff
@@ -37,49 +39,56 @@ The process of customizing the body of each message is the same as the customizi
...
 
-1. In the breadcrumb trail at the top of the Template Information section, find the following information, as it relates to your template.
+## Step 3. Update the configuration
 
-   In this example the template configuration is located on the Customer Configuration page, in the Create New Account Options section, and in the Default Welcome Email field.
+The Currently Used For breadcrumb trail at the top of the Template Information section shows where the template is used. In this example the template configuration is located on the Customer Configuration page, in the Create New Account Options section, and in the Default Welcome Email field.
```
Comment (hickskenh) [inline src/marketing/email-template-message.md:59]:
```suggestion
The Currently Used For breadcrumb trail at the top of the Template Information section shows where the template is used. In this example, the template configuration is located on the Customer Configuration page, in the Create New Account Options section, and in the Default Welcome Email field.
```

File: src/marketing/email-template-message.md [hunk 7]
```diff
@@ -37,49 +39,56 @@ The process of customizing the body of each message is the same as the customizi
...
-1. Click the link to open the template configuration page.
+1. In the **Currently Used For** breadcrumb trail, click the link to open the template configuration page.
 
     ![]({% link images/images/email-template-new-currently-used-for.png %}){: .zoom}
     _Current email template_
 
-1. Expand ![Expansion selector]({% link images/images/btn-expand.png %}) the section, find the field for the email template that you customized, and specify the new custom template as the default.
+1. Expand ![Expansion selector]({% link images/images/btn-expand.png %}) the section, and find the field for the email template that you customized.
+
+1. Clear the **Use system value** checkbox, and click the name of your custom template.
```
Comment (hickskenh) [inline src/marketing/email-template-message.md:72]:
```suggestion
1. Clear the **Use system value** checkbox and click the name of your custom template.
```

File: src/marketing/variables-custom.md [hunk 8]
```diff
@@ -4,38 +4,94 @@ redirect_from:
   - /system/variables-custom.html
 ---
 
-If you know a little basic HTML, you can create custom variables and use a [markup tag]({% link marketing/markup-tags.md -%}) to incorporate them into [email]({% link marketing/email-templates.md -%}) and [newsletter]({% link marketing/newsletter-template.md -%}) templates, as well as other types of [content]({% link cms/content-elements.md -%}).
+To meet the specific needs of your business, you can create custom variables and insert them into [pages]({% link cms/pages.md -%}), [blocks]({% link cms/blocks.md -%}) and [email]({% link marketing/email-template-message.md %}) templates. The list of whitelisted variables that appears when you click the Insert Variable button includes both [predefined]({% link marketing/variables-predefined.md %}) and custom variables. As shown in the following image, the list of available variables for a specific email template is determined by the data that is associated with the template. See the [Variable Reference]({% link marketing/variables-reference.md -%}) for a list of frequently-used email templates and their associated variables.
```
Comment (hickskenh) [inline src/marketing/variables-custom.md:7]:
```suggestion
To meet the specific needs of your business, you can create [custom]({% link marketing/variables-custom.md %}) variables and insert them into [pages]({% link cms/pages.md -%}), [blocks]({% link cms/blocks.md -%}) and [email]({% link marketing/email-template-message.md %}) templates. The list of whitelisted variables that appears when you click the Insert Variable button includes both [predefined]({% link marketing/variables-predefined.md %}) and [custom]({% link marketing/variables-custom.md %}) variables. As shown in the following image, the list of available variables for a specific email template is determined by the data that is associated with the template. See the [Variable Reference]({% link marketing/variables-reference.md -%}) for a list of frequently-used email templates and their associated variables.
```

File: src/marketing/variables-custom.md [hunk 9]
```diff
@@ -4,38 +4,94 @@ redirect_from:
...
 
-![Custome variables]({% link images/images/variables-custom.png %}){: .zoom}
-_Custom Variables_
+{:.bs-callout-info}
+Only whitelisted [predefined]({% link marketing/variables-predefined.md %}) or custom variables can be used in email and newsletter templates.
```
Comment (hickskenh) [inline src/marketing/variables-custom.md:10]:
```suggestion
Only whitelisted [predefined]({% link marketing/variables-predefined.md %}) or [custom]({% link marketing/variables-custom.md %}) variables can be used in email and newsletter templates.
```

File: src/marketing/variables-custom.md [hunk 10]
```diff
@@ -4,38 +4,94 @@ redirect_from:
...
+## Step 1: Create a custom variable
 
 1. On the _Admin_ sidebar, go to **System** > _Other Settings_ > **Custom Variables**.
 
 1. Click <span class="btn">Add New Variable</span>.
 
+   ![Custom variables]({% link images/images/variables-custom.png %}){: .zoom}
+   _Custom Variables_
+
 1. Enter an identifier in the **Variable Code** field, using all lowercase characters without spaces.
```
Comment (hickskenh) [inline src/marketing/variables-custom.md:21]:
```suggestion
1. Enter an identifier for **Variable Code**, using all lowercase characters without spaces.
```

File: src/marketing/variables-custom.md [hunk 11]
```diff
@@ -4,38 +4,94 @@ redirect_from:
...
-    If needed, you can use an underscore character to represent a space. For example: `my_custom_variable`
+   If needed, you can use an underscore character or hyphen to represent a space. For example: `my_custom_variable`
 
 1. Enter a **Variable Name**, which is used for internal reference. For example: `My Custom Variable`
 
 1. To enter the value that is associated with the variable, do one of the following:
 
-   - In the **Variable HTML Value** field, enter the variable value formatted with simple HTML tags. For example:
-   `<b>This formatted content appears in place of the variable.</b>`
+   - In the **Variable HTML Value** field, enter the variable value formatted with simple HTML tags For example:
```
Comment (hickskenh) [inline src/marketing/variables-custom.md:29]:
```suggestion
   - For **Variable HTML Value**, enter the variable value formatted with simple HTML tags For example:
```

File: src/marketing/variables-custom.md [hunk 12]
```diff
@@ -4,38 +4,94 @@ redirect_from:
...
+      `<b>This formatted content appears in place of the variable.</b>`
    - In the **Variable Plain Value** field, enter the variable value as plain text without formatting. For example:
```
Comment (hickskenh) [inline src/marketing/variables-custom.md:31]:
```suggestion
   - For **Variable Plain Value**, enter the variable value as plain text without formatting. For example:
```

File: src/marketing/variables-custom.md [hunk 13]
```diff
@@ -4,38 +4,94 @@ redirect_from:
...
+## Step 2: Insert the custom variable
+
+<!--{% if "Default.CE Only" contains site.edition %}-->
+The following example shows how to insert a variable into a page.
+
+1. Open the CMS page or block where the variable is to appear.
+
+1. Expand ![]({% link images/images/btn-expand.png %}){: .Inline} the **Content** section. Then, click <span class="btn">Show / Hide Editor</span> to work in HTML.
+
+1. Position the insertion point in the editor where you want the variable to appear. Then, click <span class="btn">Insert Variable</span>.
```
Comment (hickskenh) [inline src/marketing/variables-custom.md:51]:
```suggestion
1. Position the insertion point in the editor where you want the variable to appear and click <span class="btn">Insert Variable</span>.
```

File: src/marketing/variables-custom.md [hunk 14]
```diff
@@ -4,38 +4,94 @@ redirect_from:
...
+
+1. Select the option for the custom variable that you want to insert. Then, click <span class="btn">Insert Variable</span>.
```
Comment (hickskenh) [inline src/marketing/variables-custom.md:53]:
```suggestion
1. Select the option for the custom variable that you want to insert and click <span class="btn">Insert Variable</span>.
```

File: src/marketing/variables-custom.md [hunk 15]
```diff
@@ -4,38 +4,94 @@ redirect_from:
...
+   _Select Custom Variable_
+
+   A command to insert the variable is enclosed in curly braces and added to the code at the cursor location. For example:
+
+   `customVar code=my_custom_variable`
+
+   ![New custom variable]({% link images/images/variable-custom-insert-content.png %}){: .zoom}
+   _Custom Variable Code_
+
+1. When the content is complete, click <span class="btn">Save</span>.
```
Comment (hickskenh) [inline src/marketing/variables-custom.md:65]:
```suggestion
1. When complete, click <span class="btn">Save</span>.
```

File: src/marketing/variables-custom.md [hunk 16]
```diff
@@ -4,38 +4,94 @@ redirect_from:
...
+<!--{% endif %}-->
+
+<!--{% if "Default.EE-B2B" contains site.edition %}-->
+The following example shows how to use [Page Builder]({% link cms/page-builder.md %}) to insert a custom variable into a page.
+
+1. Open the page or block where the variable is to appear.
+
+1. Expand ![]({% link images/images/btn-expand.png %}){: .Inline} the **Content** section.
+
+1. In the left panel, click **Elements**. Then, do one of the following:
```
Comment (hickskenh) [inline src/marketing/variables-custom.md:75]:
```suggestion
1. In the left panel, click **Elements** and do one of the following:
```

File: src/marketing/variables-custom.md [hunk 17]
```diff
@@ -4,38 +4,94 @@ redirect_from:
...
+   - Click in an existing text area where you want to insert the variable.
+
+   - Drag a new **Text** object to the stage.
+
+1. At the far right of the editor toolbar, click (![]({% link images/images/editor4-toolbar/btn-insert-variable.png %}){: .Inline}) to insert a variable.
+
+   ![Page Builder stage]({% link images/images-ee/variable-custom-pagebuilder-stage.png %}){: .zoom}
+   _PageBuilder Stage and Panel_
+
+1. In the list, select the custom variable that you want to insert. Then, click <span class="btn">Insert Variable</span>.
```
Comment (hickskenh) [inline src/marketing/variables-custom.md:86]:
```suggestion
1. In the list, select the custom variable that you want to insert and click <span class="btn">Insert Variable</span>.
```

File: src/marketing/variables-custom.md [hunk 18]
```diff
@@ -4,38 +4,94 @@ redirect_from:
...
+   ![New custom variable]({% link images/images-ee/variable-custom-insert-select.png %}){: .zoom}
+   _Select Custom Variable_
+
+   The variable identifier appears as a placeholder in the editor.
+
+   ![Page Builder stage]({% link images/images-ee/pagebuilder-variable-inserted.png %}){: .zoom}
+   _Variable Placeholder_
 
-  'CustomVar code= "my_custom_variable"'
+1. When the content is complete, click <span class="btn">Save</span>.
```
Comment (hickskenh) [inline src/marketing/variables-custom.md:96]:
```suggestion
1. When complete, click <span class="btn">Save</span>.
```

File: src/marketing/variables-predefined.md [hunk 19]
```diff
@@ -2,24 +2,39 @@
...
+Predefined variables make it easy to personalize [email]({% link marketing/email-templates.md -%}) and [newsletter]({% link marketing/newsletters.md -%}) templates, as well as other types of [content]({% link cms/content-elements.md -%}). The list of whitelisted predefined variables appears when you click the Insert Variable button. As shown in the following image, the list of available variables for a specific email template is determined by the data that is associated with the template. See the [Variable Reference]({% link marketing/variables-reference.md -%}) for a list of frequently-used email templates and their associated variables.
```
Comment (hickskenh) [inline src/marketing/variables-predefined.md:5]:
```suggestion
[Predefined]({% link marketing/variables-predefined.md %}) variables make it easy to personalize [email]({% link marketing/email-templates.md -%}) and [newsletter]({% link marketing/newsletters.md -%}) templates, as well as other types of [content]({% link cms/content-elements.md -%}). The list of whitelisted [predefined]({% link marketing/variables-predefined.md %}) variables appears when you click the Insert Variable button. As shown in the following image, the list of available variables for a specific email template is determined by the data that is associated with the template. See the [Variable Reference]({% link marketing/variables-reference.md -%}) for a list of frequently-used email templates and their associated variables.
```

File: src/marketing/variables-predefined.md [hunk 20]
```diff
@@ -2,24 +2,39 @@
...
-## Add a Variable to an Email Template
+## Add a variable to an email template
 
 1. On the _Admin_ sidebar, go to **Marketing** > _Communications_ > **Email Templates**.
 
 1. Do one of the following:
 
     - To add the variable to an existing template, click the template in the list to open in edit mode.
+
     - To use the variable in a new template, click <span class="btn">Add New Template</span> and customize the default template code. To learn more, see [Message Templates]({% link marketing/email-template-message.md %}).
```
Comment (hickskenh) [inline src/marketing/variables-predefined.md:18]:
```suggestion
    - To use the variable in a new template, click <span class="btn">Add New Template</span> and customize the default template code. See [Message Templates]({% link marketing/email-template-message.md %}).
```

File: src/marketing/variables-predefined.md [hunk 21]
```diff
@@ -2,24 +2,39 @@
...
 
+1. For a list of the predefined variables that can be used with this email template, click <span class="btn">Insert Variable</span>.
```
Comment (hickskenh) [inline src/marketing/variables-predefined.md:20]:
```suggestion
1. For a list of the [predefined]({% link marketing/variables-predefined.md %}) variables that can be used with this email template, click <span class="btn">Insert Variable</span>.
```

File: src/marketing/variables-custom.md (hunk 9 above)
Comment (ccondra) [reply src/marketing/variables-custom.md:10]: This is OK as is -- it's not necessary to link to itself.

File: src/marketing/variables-custom.md (hunk 8 above)
Comment (ccondra) [reply src/marketing/variables-custom.md:7]: This is OK as is -- it's not necessary to link to itself.

File: src/marketing/variables-custom.md (hunk 11 above)
Comment (ccondra) [reply src/marketing/variables-custom.md:29]: Fixed. (There was also a missing period)

File: src/marketing/variables-custom.md (hunk 12 above)
Comment (ccondra) [reply src/marketing/variables-custom.md:31]: Fixed. (committed along with previous one)

**Comment by ccondra** [comment]
running tests

**Pull request merged**